.buildkite/steps/test-demo-notebooks.sh

```diff
@@ -18,7 +18,14 @@ done < <(find "${stellargraph_dir}/demos" -name "*.ipynb" -print0)
 NUM_NOTEBOOKS_TOTAL=${#NOTEBOOKS[@]}
 
 if [ "$NUM_NOTEBOOKS_TOTAL" -ne "$SPLIT" ]; then
-  echo "Buildkite step parallelism is set to $SPLIT but found $NUM_NOTEBOOKS_TOTAL notebooks. Please update parallelism to match the number of notebooks!"
+  msg="Buildkite step parallelism is set to $SPLIT but found $NUM_NOTEBOOKS_TOTAL notebooks. Please update the notebook testing \`parallelism\` setting in \`.buildkite/pipeline.yml\`  to match the number of notebooks!
+
+If a pull request adding or removing notebooks was merged recently, this may be fixed by doing a merge with \`develop\`; for example: \`git fetch origin && git merge origin/develop\`."
```

Comment:
just to clarify, when will merging with develop fix this?

Reply:
When a PR adding or removing notebooks was merged into `develop`, and the current build's commit doesn't inherit from that merged commit, e.g.: if the git DAG for `develop` and the PR in question looks like

```
D (develop)
|
|  P (PR)
| /
B
|
older
|
commits
|
...
```

Suppose commit `D` on `develop` adds a notebook and so has `parallelism: 40`, vs. the PR commit `P` that has `parallelism: 39`. When `P` is built against `D`, the `pipeline.yml` will be uploaded from `P` (and so `parallelism: 39`). The notebook steps will be run after doing a transient merge with `D` on the CI machines, the `plugins` anchor pulls in the https://github.com/stellargraph/github-merged-pr-buildkite-plugin plugin:


https://github.com/stellargraph/stellargraph/blob/913cf976dbfadd0320a16535e0285fe3424f8572/.buildkite/pipeline.yml#L82-L89



After doing the merge with `D`, the code being tested for `P` will have 40 notebooks (the merge will bring in that extra notebook), and thus the parallelism setting of 39 will be wrong. If an explicit merge is done and pushed to the PR, the parallelism in the `pipeline.yml` will be correct (40):

```
   M (PR)
 / |
D (develop)
|  | 
|  P
| /
B
|
|
older
|
commits
|
...

```

Reply:
ah makes sense, I wasn't aware of the transient merge going on that would pull in extra notebooks from develop 👍